autofit/graphical/messages/abstract.py

```diff
@@ -1,17 +1,20 @@
 from abc import ABC, abstractmethod
-from functools import reduce
+from functools import reduce, lru_cache
 from itertools import chain
 from operator import and_
-from typing import Optional, Tuple, Union, Iterator
+from typing import Optional, Tuple, Union, Iterator, Type, List
 from inspect import getfullargspec
 
 import numpy as np
 
-from autofit.mapper.variable import Variable
-
+from ...mapper.variable import Variable
+from ..factor_graphs.jacobians import FactorJacobian
+from ..utils import cached_property, jac_grad_betaln
+from .transform import AbstractDensityTransform, LinearShiftTransform
 
 class AbstractMessage(ABC):
     log_base_measure: float
+    _projection_class: Optional[Type['AbstractMessage']] = None
```

Comment:
The property needs to be accessible from classmethods. 

I had added the method in case there was a special case of a AbstractMessage that needed to be handled differently, e.g. Haldane's prior for the Beta Distribution, but actually atm there might be no need for it.

Reply:
If it's not necessary yet I'd be inclined to remove it and rely on accessing the class from inside a method as it reduces code complexity